idaes/ui/flowsheet.py

```diff
@@ -292,6 +294,7 @@ def _add_unit_model_with_ports(self, unit):
                 performance_df["Value"] = performance_df["Value"].map(
                     lambda v: value(v)
                 )
+                performance_df = performance_df.replace(np.nan, "NaN", regex=True)
```

Comment:
Since this conversion step is applied more than once (and, more importantly, for the same purpose), we could consider extracting it to an utility function, e.g. `_convert_for_ajax_compatibility(df: pd.DataFrame) -> pd.DataFrame`

Reply:
Good call. Done.